src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -279,86 +472,40 @@ public void ProcessPushNotification(PushNotification pushNotification, TimeSpan?
                     $"{nameof(pushNotification)}.{nameof(pushNotification.ResourceUri)}");
             }
 
-            _client.UpdateSyncToken(pushNotification.SyncToken);
-
-            SetDirty(maxDelay);
+            if (_configClientManager.UpdateSyncToken(pushNotification.ResourceUri, pushNotification.SyncToken))
+            {
+                SetDirty(maxDelay);
+            }
+            else
+            {
+                _logger.LogWarning($"Ignoring the push notification received for the unregistered endpoint '{pushNotification.ResourceUri}'");
+            }
         }
 
-        private async Task LoadAll(bool ignoreFailures, CancellationToken cancellationToken)
+        private async Task InitializeAsync(bool ignoreFailures, CancellationToken cancellationToken = default)
         {
-            IDictionary<string, ConfigurationSetting> data = null;
-            bool success = false;
+            Dictionary<string, ConfigurationSetting> data = null;
 
             try
             {
-                var serverData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
-
-                // Use default query if there are no key-values specified for use other than the feature flags
-                bool useDefaultQuery = !_options.KeyValueSelectors.Any(selector => !selector.KeyFilter.StartsWith(FeatureManagementConstants.FeatureFlagMarker));
-
-                if (useDefaultQuery)
+                IEnumerable<ConfigurationClient> availableClients = _configClientManager.GetAvailableClients();
+                if (!availableClients.Any())
                 {

```

Comment:
Doesn't need to be in the try block since it doesn't throw. Also, it seems off that we check this in the `Load` code path. And in refresh async code path we end up getting the clients twice. Did you consider to pass clients to this helper method?

Reply:
Yea, that is a good idea!
Done!